core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -216,7 +216,7 @@ public static ProxyConfiguration load() throws IOException {
      */
     public static URLConnection open(URL url) throws IOException {
         Jenkins h = Jenkins.getInstanceOrNull(); // this code might run on slaves
```

Comment:
Nope `Channel.current` is only valid during serialization / deserialization.

We really need a helper API to confirm if we are on the master JVM